test/extended/openstack/machinesets.go

```diff
@@ -0,0 +1,135 @@
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
+
+	defer g.GinkgoRecover()
+
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+
+	g.Context("after deletion of a machineset", func() {
+		g.BeforeEach(func() {
+			g.By("preparing openshift dynamic client")
+			cfg, err := e2e.LoadConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			dc, err = dynamic.NewForConfig(cfg)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			clientSet, err = e2e.LoadClientset()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("preparing openstack client")
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+		})
+
+		g.It("should not have leftovers ports", func() {
+
+			skipUnlessMachineAPIOperator(dc, clientSet.CoreV1().Namespaces())
+			g.By("fetching worker machineSets")
+			var networkClient *gophercloud.ServiceClient
+			var rawBytes []byte
+			var t machineproviderv1.OpenstackProviderSpec
+			var rclient runtimeclient.Client
+
+			networkClient, err := client(serviceNetwork)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			machineSets, err := listWorkerMachineSets(dc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			if len(machineSets) == 0 {
+				e2eskipper.Skipf("Expects at least one worker machineset. Found none.")
+			}
+			xconfig, err := config.GetConfig()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			rclient, err = runtimeclient.New(xconfig, runtimeclient.Options{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = machinev1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			err = configv1.AddToScheme(scheme.Scheme)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			newMachinesetParams := framework.BuildMachineSetParams(rclient, 1)
+			rawBytes, err = json.Marshal(newMachinesetParams.ProviderSpec.Value)
+
+			err = json.Unmarshal(rawBytes, &t)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			t.ServerGroupName = ""
+			t.ServerGroupID = "boo"
+			newMachinesetParams.Name = "bogus1-worker"
+			s, err := json.Marshal(t)
+			newMachinesetParams.ProviderSpec.Value.Raw = s
+
+			o.Expect(err).NotTo(o.HaveOccurred())
+			ms, err := framework.CreateMachineSet(rclient, newMachinesetParams)
+
+			clusterID := ms.Spec.Template.ObjectMeta.Labels["machine.openshift.io/cluster-api-cluster"]
+
+			_, err = framework.GetMachineSet(rclient, ms.Name)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			//Delete the machineset
+			e2e.Logf("Waiting for 10 seconds before deleting machineset %v", ms.Name)
```

Comment:
What happens if 10 seconds was not enough to create the machine? Perhaps we should poll instead